test/functional/test_framework/ellsq.py

```diff
@@ -5,64 +5,93 @@
 keys, and is trivially vulnerable to side channel attacks. Do not use for
 anything but tests."""
 
+class fe:
+    """Prime field over 2^256 - 2^32 - 977"""
+    p = 2**256 - 2**32 - 977
+    def __init__(self, x):
+        self.val = x % self.p
+
+    def __add__     (self, o): return fe( self.val +  o.val                   )
+    def __eq__      (self, o): return self.val % self.p == o.val % self.p
```

Comment:
I don't think the modulo reductions are strictly needed; any `fe` object's `val` value should already be reduced.

Reply:
Interesting! Done.